File: golang/19_remove_nth_node_from_end_of_list.go
package main

// ListNode Definition for singly-linked list.
type ListNode struct {
	Val  int
	Next *ListNode
}

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	temp := &ListNode{
		Val:  -1,
		Next: head,
	}

	fastPtr := temp
	slowPtr := temp

	for i := 0; i < n+1; i++ {
		fastPtr = fastPtr.Next
	}

	for fastPtr != nil {
		fastPtr = fastPtr.Next
		slowPtr = slowPtr.Next
	}
	slowPtr.Next = slowPtr.Next.Next
	return temp.Next
}

File: python/19_remove_nth_node_from_end_of_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        temp = ListNode(-1)
        temp.next = head
        
        fast_ptr = temp
        slow_ptr = temp
        
        for i in range(n + 1):
            fast_ptr = fast_ptr.next
        
        while fast_ptr:
            fast_ptr = fast_ptr.next
            slow_ptr = slow_ptr.next
        
        slow_ptr.next = slow_ptr.next.next
        return temp.next